### PR TITLE
Disable create game button when incompatible optional rules are selected

### PR DESCRIPTION
**Pull request by michaeljb**
Old behavior with bad set of optional rules selected:

* multiplayer: can create a game table, no error until starting the game
* hotseat: taken to error page
* JSON: taken to error page

New behavior with bad set:

* multiplayer and hotseat: 'Create' button is disabled
* JSON: error message flashes and user stays on the page

### CODE REVIEW

File: assets/app/view/create_game.rb [hunk 1]
```diff
@@ -408,6 +421,13 @@ def submit
         game_data[:settings][:seed] = game_params[:seed] if game_params[:seed]
       end
 
+      checked_options = Engine.meta_by_title(game_data[:title])
+                          .check_options(game_data[:settings][:optional_rules],
+                                         game_data[:min_players], game_data[:max_players])
```
Comment (roseundy) [inline assets/app/view/create_game.rb:426]: Why doesn't this need a .respond_to? guard clause?

Comment (michaeljb) [reply assets/app/view/create_game.rb:424]: On the branch where I'm fixing the 1822PNW variants I added it to the base meta class, I tried splitting it out from this branch but that should be here too. And I guess with that method I can get rid of the other `.respond_to(:check_options)` here. What do you think? b89fe89

Comment (roseundy) [reply assets/app/view/create_game.rb:424]: Well, if check_options is now in the base class, then you should definitely get rid of all the repond_to guards.

**Review by roseundy — APPROVED** [review]
Looks good to me

**Pull request merged**